pkg/config/msp.go

```diff
@@ -74,3 +100,63 @@ func addRootCAToOrgMSP(rootCA *x509.Certificate, org *cb.ConfigGroup) error {
 
 	return nil
 }
+
+func getOrgMSPValue(org *cb.ConfigGroup) (*cb.ConfigValue, error) {
+	configValue, ok := org.Values[MSPKey]
+	if !ok {
+		return nil, errors.New("org doesn't have MSP value")
+	}
+	return configValue, nil
+}
+
+// RevokeCertificateFromMSP takes an x509 certificate and adds it to the
+// revocation list for the specified application org MSP.
+func RevokeCertificateFromMSP(cert *x509.Certificate, orgName string, config *cb.Config) error {
```

Comment:
To revoke a certificate, you need the CA that issued the certificate to sign the CRL that revokes it. 
Even under the assumption that the function as it is now, correctly constructs the CRL and correctly adds it to the MSP config - won't make the certificate be actually revoked, because you're not signing the revocation with the issuer's private key.

Reply:
Does the CRL need to be signed each time a new certificate is added? The signing is easily handled by x509.CreateCRL (which we plan to incorporate when we implement the TODO to create a CRL if one doesn't already exist) so I was under the assumption that the previous signature on an existing CRL would still be valid.

Reply:
> Does the CRL need to be signed each time a new certificate is added? 

The stuff that is signed is the TBSCertificateList which contains the SNs of the revoked certificates. If you add another certificate then you obviously need to re-sign, otherwise the validation won't work.

My main point was though, that you need to somehow pass a private key to this method...